bdns/dns.go

```diff
@@ -469,12 +469,12 @@ func (dnsClient *DNSClientImpl) LookupCAA(ctx context.Context, hostname string)
 	var CAAs []*dns.CAA
 	var CNAMEs []*dns.CNAME
 	for _, answer := range r.Answer {
-		if caaR, ok := answer.(*dns.CAA); ok {
-			CAAs = append(CAAs, caaR)
-		} else if _, ok := answer.(*dns.DNAME); ok {
-			return nil, nil, fmt.Errorf("Got DNAME when looking up CNAME. DNAMEs not supported.")
-		} else if cnameR, ok := answer.(*dns.CNAME); ok {
-			CNAMEs = append(CNAMEs, cnameR)
+		if answer.Header().Rrtype == dnsType {
```

Comment:
I'm pretty sure this breaks the CNAME return, a record cannot have `Rrtype == dns.CAA` and be casted to `*dns.CNAME`.

Reply:
I also notice we don't have a test that checks this behavior anywhere?

Reply:
I just noticed this is a straight revert, so this was probably broken before as well and we just never noticed?

Reply:
Yep, you're right. From the original PR:

> Also, in the process, fix a bug in the previous LegacyCAA implementation.
> Because the processing of records in LookupCAA was gated by `if
> answer.Header().RRType == dnsType`, non-CAA responses were filtered out. This
> wasn't caught by previous testing, because it was unittesting that mocked out
> bdns.

I'll change this to preserve that fix.

Reply:
Good catch @rolandshoemaker !